### PR TITLE
Modular compliance rules table

### PR DESCRIPTION
**Pull request by dLobatog**
These changes are required for https://github.com/RedHatInsights/compliance-frontend/pull/197

### CODE REVIEW

File: packages/inventory-compliance/src/index.js [hunk 1]
```diff
@@ -1 +1,2 @@
 export { default as default } from './Compliance';
+export { default as SystemRulesTable } from './SystemRulesTable';
```
Comment (dLobatog) [inline packages/inventory-compliance/src/index.js]: @akofink the line above was `export { Compliance as default } ` before I changed it, which is what caused the problem you saw. Since compliance-frontend uses the default export, it was failing to pick up any as it was undefined.

File: packages/inventory-compliance/src/SystemRulesTable.js [hunk 2]
```diff
@@ -211,28 +180,31 @@ class SystemRulesTable extends React.Component {
...
-                original: remediationAvailable
-            }
-        ]
-    })
+    calculateParent = ({ profile }, { title, severity, compliant, identifier, remediationAvailable }) => {
+        const { columns } = this.state;
+        return {
+            isOpen: false,
+            cells: [
+                columns.find((column) => column.title === 'Rule') ? { title: this.ruleTitleCell(title, identifier), original: title } : null,
```
Comment (bastilian) [inline packages/inventory-compliance/src/SystemRulesTable.js]: `columns.find(...) ? ... : null` could be extracted into a function.

File: packages/inventory-compliance/src/Compliance.js [hunk 3]
```diff
@@ -57,12 +59,21 @@ query System($systemId: String!){
 }
 `;
 
+const columns = [
```
Comment (bastilian) [inline packages/inventory-compliance/src/Compliance.js]: It wouldn't be the worst to export this to use in the tests, but just a suggestion.

File: packages/inventory-compliance/src/SystemRulesTable.test.js [hunk 4]
```diff
@@ -117,14 +145,15 @@ describe('SystemRulesTable component', () => {
...
+                columns={ columns }
             />
         );
         const instance = wrapper.instance();
         await instance.setInitialCurrentRows();
         await instance.updateFilter(false, [ 'high', 'medium' ], []);
         expect(wrapper.state('currentRows').length / 2).toEqual(50);
         wrapper.state('currentRows').forEach((row, i) => {
-            if (row.hasOwnProperty('parent')) {
+            if (Object.prototype.hasOwnProperty.call(row, 'parent')) {
```
Comment (bastilian) [inline packages/inventory-compliance/src/SystemRulesTable.test.js]: Curious why this change is needed? calling `hasOwnProperty` on `row` should be the same.

File: packages/inventory-compliance/src/SystemRulesTable.js [hunk 5]
```diff
@@ -559,8 +533,8 @@ class SystemRulesTable extends React.Component {
                         onCollapse={ this.onCollapse }
                         onSort={ this.onSort }
                         sortBy={ sortBy }
-                        onSelect={ (currentRows.length !== 0) ? this.onSelect : () => null }
-                        rows={ (currentRows.length === 0) ? emptyRows : currentRows }>
+                        onSelect={ (remediationsEnabled && currentRows.length !== 0) ? this.onSelect : undefined }
```
Comment (bastilian) [inline packages/inventory-compliance/src/SystemRulesTable.js]: Passing undefined could cause issues. The `Table` component could still try to call what is passed to `onSelect` and result in an unwanted error or warning. Using the mute function is not optimal. Optimal would be to only pass the prop if it's set, that would require to prepare the props before hand and spread them in.

File: packages/inventory-compliance/src/SystemRulesTable.js [hunk 6]
```diff
@@ -67,13 +42,7 @@ class SystemRulesTable extends React.Component {
...
         super(props);
         this.state = {
-            columns: [
-                { title: 'Rule', transforms: [ sortable ] },
-                { title: 'Policy', transforms: [ sortable ] },
-                { title: 'Severity', transforms: [ sortable ] },
-                { title: 'Passed', transforms: [ sortable ] },
-                { title: <React.Fragment>{ ANSIBLE_ICON } Ansible</React.Fragment>, transforms: [ sortable ] }
-            ],
+            columns: props.columns,
```
Comment (bastilian) [inline packages/inventory-compliance/src/SystemRulesTable.js]: The `columns` prop seems to be missing a type validation in the prop-types. It might be good to verify the shape of each `column` item as well. and possibly a default value.

File: packages/inventory-compliance/src/SystemRulesTable.test.js [hunk 7]
```diff
@@ -12,8 +12,17 @@ debounce.mockImplementation(fn => fn);
 global.fetch = require('jest-fetch-mock');
 
 describe('SystemRulesTable component', () => {
+    let columns;
```
Comment (bastilian) [inline packages/inventory-compliance/src/SystemRulesTable.test.js]: It doesn't really change through out the tests, why not define it with `const`?

File: packages/inventory-compliance/src/SystemRulesTable.test.js (hunk 4 above)
Comment (dLobatog) [reply packages/inventory-compliance/src/SystemRulesTable.test.js]: New `eslint` requirement for the whole project :man_shrugging:

**Comment by dLobatog** [comment]
@akofink I didn't check with `LOCAL_CHROME=false` but this PR isn't supposed to change how the systems table works (other than passing in the column headers as props) - it's meant mostly to make the rules table without remediations available in the Wizard

**Comment by dLobatog** [comment]
@akofink Just checked and it doesn't work for me either, there seems to be some problem with `withRouter` - that's on SystemDetails though, on `compliance-frontend` , not here :disappointed_relieved:  - this PR doesn't have much to do with it as without it it throws the same error.

**Comment by akofink** [comment]
Okay, yeah - I agree this PR doesn't seem to make the problem better or worse. I'm glad it's not just me! :octopus:

File: packages/inventory-compliance/src/Compliance.js [hunk 8]
```diff
@@ -144,6 +146,13 @@ SystemDetails.propTypes = {
             inventoryId: propTypes.string
         })
     }),
+    columns: propTypes.shape([
+        {
+            title: propTypes.oneOfType([ propTypes.string, propTypes.object ]).isRequired,
```
Comment (bastilian) [inline packages/inventory-compliance/src/Compliance.js]: If the title should also accept React elements you could use `PropTypes.element`, instead of the `propTypes.oneOfType(...)`, which allows (i think both) strings and objects/elements.
